### DISCUSSION TITLE
Winlogbeat Transfer Slows Down

### DISCUSSION THREAD

**Discussion by branden999** [Security-Onion-Solutions/securityonion#3023] [Unsupported Versions] [answered]
So I am currently running a Windows Event Collector on a physical Windows Server 2019 box and using WinLogBeat 7.11 to send forwarded events to a SO server running 2.3.21 (Logstash 7.9.3). My previous configuration (prior to yesterday) was a virtual Windows Event Collector running Windows Server 2016 and using WinLogBeat 6.8.13 to send forwarded events to a SO server running 16.x (Logstash 6.7.2). So I moved from a virtual WEC to physical ("older" server hardware, but should have plenty of resources). The SO server is physical (newer server purchased within the past 2 years and has plenty of resources). Yesterday I completely formatted the SO server and started an entirely new SO 2.3.21 installation as the previous 16.x installation broke upon running SOUP to try to update. Onto the problem...

When I start the WinLogBeat service on the WEC, events seem to flood into SO for a period of time, but then transfer slows way down and events only trickle in. Generally, if I restart the WinLogBeat service on the WEC the events will again begin to flood in, but once again taper off. Some events continue to trickle in after the "flood", but it is a very small fraction of the total number of events. All the while, the events are making it into the WEC via WEF from a handful of servers. I experienced this same issue prior to upgrading to WinLogBeat 7.11 and SO 2.3.21 also. There is no external firewall between the WEC and SO servers and no other device that would be rate-limiting the connection (layer 3/routed internal network). Both servers have a 1GB connection.

Has anyone experienced this or have any ideas where I can begin looking? I haven't been able to find anything in the WinLogBeat nor Logstash logs as of yet.

Thanks,

Branden

**Comment by weslambert** [comment, marked as answer]
What is the approximate EPS (Events per second), or throughput for the traffic being monitored?  You might first try increasing the `maxmemory` setting in `/opt/so/saltstack/local/pillar/global.sls`  for Redis (and restart with `so-redis-restart`) to see if that helps at all.

**Reply by branden999** [reply]
What is the best way to determine EPS? This SO server should be receiving events from ~2k workstations, and a small number of servers including our DCs. I can't recall exactly and since I had to reload the server Wednesday to get it to 2.3.21 I have no data in front of me, but on a good day I'm sure it was doing 10's of millions of events per day. Looks like the interface traffic is currently a measly 100k or less (1GB connections on both ends and possibly even on the same L2 switch, but I'd have to verify that). REDIS_MAXMEMORY was set to the default 812, so I bumped it to 8192 and restarted REDIS. THE SO server has 196GB of RAM and its only job in life is SO, so shouldn't be an issue spreading the wealth a bit. For that matter, I have 196GB of memory in my WEC too which I am partially using for a RAM drive to store the Event Logs on until they are shipped off to SO. 

Thanks,

Branden

**Reply by branden999** [reply]
Bumping the REDIS_MAXMEMORY doesn't seem to have helped. I still have the same issue in that WinLogBeat seems to flood LogStash with forwarded events immediately after the service is started, but then after a few moments the of events is reduced dramatically. For whatever reason, it seems like WinLogBeat either stops reading events or loses its place. I've been trying to play around with the WinLogBeat config to see if anything helps, but nothing has so far. Do you have any other thoughts or recommendations?

Thanks,

Branden

**Reply by branden999** [reply]
UPDATE: Whatever is going on seems to be with WinLogBeat or Windows Server 2019, not Logstash or anything in between. I changed the output of WinLogBeat this morning to a local file and got the same result, it fills 1 100MB file pretty quickly (within seconds) and then takes another several hours to fill another even though events are still rolling in via WEF to the event log the whole time.

**Reply by weslambert** [reply]
You might want to try adding a `worker` parameter to the config file to see if it works better for you:
https://www.elastic.co/guide/en/beats/winlogbeat/current/logstash-output.html#_worker_2

**Reply by branden999** [reply]
I did bump the number of workers up to 2 early last week, didn't seem to make much of a difference. I'm attaching a sanitized copy of my current WinLogBeat for you to look at to see if you notice anything wrong:

[winlogbeat-REDACTED.txt](https://github.com/Security-Onion-Solutions/securityonion/files/6062558/winlogbeat-REDACTED.txt)

**Reply by branden999** [reply]
Does anyone have any tips, tricks, or learning experiences to share with getting WinLogBeat to run on Windows Server 2019?

Branden

**Reply by defensivedepth** [reply]
Couple thoughts:

- If you haven't already, I would lower your logging level in WLB - you should be seeing some type of log messages that describe what is going on.

- Check out this WLB docs page - There are a few config options here to try and tweak: https://www.elastic.co/guide/en/beats/winlogbeat/current/configuring-internal-queue.html#configuring-internal-queue

- Finally - If it is a WLB issue, you could try posting on the Elastic forums for further help

**Reply by branden999** [reply]
Thank you for the suggestions. I had already been logging debug output for WLB, but that wasn't very fruitful. I had already tried a number of the tweaks in that reference as well, again nothing seemed to make a difference. Also, I had a post on Elastic's forum running parallel to this one, but it did not receive much attention either. Ultimately, I fought with Windows Server 2019 too long and didn't have much progress to show for it. So in order to get our logging back up I just finished rebuilding both of our WECs on Windows Server 2016 and am having much better success. New version of WLB (7.12.0 which was released on 03/22), but the exact same config. No other tweaks to speak of. I guess I'll have to fight the Windows Server 2019 battle in my lab environment.

Thanks again,

Branden

**Reply by defensivedepth** [reply]
@branden999 Ok good luck! Keep fighting the good fight! :)
